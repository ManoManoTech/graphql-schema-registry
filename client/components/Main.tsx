import { Route, Redirect, Switch } from 'react-router-dom';

import TopMenu from './TopMenu';
import Schema from '../schema';
import PersistedQueries from '../persisted-queries';
import Clients from '../clients';

import ServicesTab from '../schema/Tab';
import PersistedQueriesTab from '../persisted-queries/Tab';
import Logs from '../logs';
import Types from '../types';
import styled from 'styled-components';
import { DateRangeSelectorProvider } from './DateRangeSelector.Context';

const MainContainer = styled.main`
	height: 100vh;
	display: grid;
	grid-template-rows: auto 1fr;
`;

export const tabs = [
	{
		title: <ServicesTab />,
		href: '/schema',
		component: Schema,
	},
	{
		title: <span>Types</span>,
		href: '/types',
		component: Types,
	},
	{
		title: <PersistedQueriesTab />,
		href: '/persisted-queries',
		component: PersistedQueries,
	},
	{
		title: <span>Clients</span>,
		href: '/clients',
		component: Clients,
	},
	{
		title: <span>Logs</span>,
		href: '/logs',
		component: Logs,
	},
];

const Main = () => (
	<MainContainer>
		<DateRangeSelectorProvider>
			<TopMenu tabs={tabs} />
			<Switch>
				{tabs.map(({ component: Component, href }) => (
					<Route key={href} path={href}>
						<Component />
					</Route>
				))}
				<Redirect to={tabs[0].href} />
			</Switch>
		</DateRangeSelectorProvider>
	</MainContainer>
);

export default Main;
