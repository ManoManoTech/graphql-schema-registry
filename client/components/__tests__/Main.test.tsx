import { render, screen } from '@testing-library/react';
import { MemoryRouter, MemoryRouterProps } from 'react-router-dom';
import Main, { tabs } from '../Main';

jest.mock('../TopMenu', () => ({
	__esModule: true,
	default: () => 'TopMenu',
}));

jest.mock('../../schema', () => ({
	__esModule: true,
	default: () => 'Schema',
}));

jest.mock('../../types', () => ({
	__esModule: true,
	default: () => 'Types',
}));

jest.mock('../../persisted-queries', () => ({
	__esModule: true,
	default: () => 'PersistedQueries',
	Tab: () => 'Tab',
}));

const customRender = (
	ui: React.ReactElement,
	customProps: Partial<MemoryRouterProps>
) => {
	return render(ui, {
		wrapper: (props: MemoryRouterProps) => (
			<MemoryRouter {...props} {...customProps} />
		),
	});
};

describe('Main tests', () => {
	it('should render for Schema', async () => {
		customRender(<Main />, {
			initialEntries: [tabs[0].href],
			initialIndex: 0,
		});
		const main = await screen.findByRole('main');

		expect(main).toBeInTheDocument();
		expect(main).toMatchSnapshot();
	});

	it('should render for Types', async () => {
		customRender(<Main />, {
			initialEntries: [tabs[1].href],
			initialIndex: 1,
		});

		const main = await screen.findByRole('main');

		expect(main).toBeInTheDocument();
		expect(main).toMatchSnapshot();
	});

	it('should render for PersistedQueries', async () => {
		customRender(<Main />, {
			initialEntries: [tabs[2].href],
			initialIndex: 2,
		});

		const main = await screen.findByRole('main');

		expect(main).toBeInTheDocument();
		expect(main).toMatchSnapshot();
	});
});
