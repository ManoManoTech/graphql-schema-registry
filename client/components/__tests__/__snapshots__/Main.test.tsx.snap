// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main tests should render for PersistedQueries 1`] = `
<main
  class="Main__MainContainer-sc-sjcz9g-0 bSxpxG"
>
  TopMenu
  PersistedQueries
</main>
`;

exports[`Main tests should render for Schema 1`] = `
<main
  class="Main__MainContainer-sc-sjcz9g-0 bSxpxG"
>
  TopMenu
  Schema
</main>
`;

exports[`Main tests should render for Types 1`] = `
<main
  class="Main__MainContainer-sc-sjcz9g-0 bSxpxG"
>
  TopMenu
  Types
</main>
`;
